ignite/metrics/ssim.py

```diff
@@ -0,0 +1,178 @@
+from typing import Callable, Sequence
+
+import torch
+import torch.nn.functional as F
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["SSIM"]
+
+
+class SSIM(Metric):
+    """
+    Computes Structual Similarity Index Measure
+
+    Args:
+        kernel_size (list or tuple of int): Size of the gaussian kernel. Default: (11, 11)
+        sigma: Standard deviation of the gaussian kernel. Default: (1.5, 1.5)
```

Comment:
@ydcjeff I wanted to say that all arguments: sigma, data_range etc need to be annotated with the types like for `kernel_size`. Thanks !